code_map.py

```diff
@@ -312,69 +363,129 @@ def run(self, edit, **args):
 
         def do():
-            get_code_map_view().run_command('scroll_to_left')
+            get_code_map_view().run_command('code_map_scroll_to_left')
+
+        sublime.set_timeout(do, 10)
 
-        sublime.set_timeout(do, 100)
     # -----------------
+
     def run(self, edit):
         region = self.view.visible_region()
-        y = self.view.text_to_layout(region.begin())[1]
+        y = self.view.text_to_layout(region.a)[1]
         self.view.set_viewport_position((0, y), False)
 
-# ===============================================================================
-class synch_code_map(sublime_plugin.TextCommand):
-    # -----------------
+
+class code_map_increase_depth(sublime_plugin.TextCommand):
+
     def run(self, edit):
-        map_view = get_code_map_view()
 
-        if map_view and len(map_view.sel()) > 0:
-            code_view_line, _ = self.view.rowcol(self.view.sel()[0].begin())
+        file = self.view.file_name()
+        d = settings().get('depth')
+        fD = Mapper.DEPTH[1]
 
-            prev_member_line_num = 0
-            prev_map_line = None
+        if file in fD and fD[file] < 4:
+            fD[file] += 1
+        elif d < 4:
+            fD[file] = d + 1
 
-            lines = map_view.lines(sublime.Region(0, map_view.size()))
+        refresh_map_for(self.view)
+        synch_map(self.view)
 
-            for line in lines:
-                link = map_view.substr(line).split(':')[-1]
 
-                member_line_num = None
-                try:
-                    member_line_num = int(link)
-                except:
-                    continue
+class code_map_decrease_depth(sublime_plugin.TextCommand):
 
-                if member_line_num and member_line_num > code_view_line:
-                    break
+    def run(self, edit):
 
-                else:
-                    prev_member_line_num = member_line_num
-                    prev_map_line = line
+        file = self.view.file_name()
+        d = settings().get('depth')
+        fD = Mapper.DEPTH[1]
 
-            map_view.sel().clear()
-            map_view.sel().add(prev_map_line)
-            map_view.show_at_center(prev_map_line)
-            map_view.window().focus_view(map_view)
-            scroll_to_left.code_map_view(self.view)
+        if file in fD and fD[file] > 0:
+            fD[file] -= 1
+        elif d > 0:
+            fD[file] = d - 1
+
+        refresh_map_for(self.view)
+        synch_map(self.view)
+
+
+class navigate_code_map(sublime_plugin.TextCommand):
+
+    def run(self, edit, direction=None, start=False, stop=False, fast=False):
+
+        def scroll():
+            v = CodeMapListener.nav_view
+            line = v.line(v.sel()[0].a)
+            y = v.text_to_layout(line.a)[1]
+            v.set_viewport_position((0, y-200), False)
+
+        v = self.view
+        cm = get_code_map_view()
+        CodeMapListener.skip = True
+
+        if start:
+            if not CodeMapListener.nav_view:
```

Comment:
This has been fixed in my uncommitted changes, so that the navigable map doesn't go crazy, it should be:
`if not CodeMapListener.nav_view or not CodeMapListener.navigating:`